design-documents/graph-ql/coverage/customer/wishlist.md

```diff
@@ -111,3 +111,35 @@ type WishlistItem {
 }
 ```
 This value was previously used for display only, other operations like update or delete are not implemented yet.
+
+## Exposure through product interface
+
+Products need to know which wishlists are they assigned to. Just like categories
+
+``` graphql
+type ProductInterface {
+    wishlists: [Wishlist]! @doc(description: "A product can be assigned to multiple wishlist of none")
+}
+```
+
+By default a product is not assigned to any wishlist
+
+### Considerations of performance versus graphql specs
+The UI really needs this to render a PDP page or Category with products page and list a dropdown of wishlists of which the product is assigned to.
```

Comment:
@supernova-at how do you feel about this? we would lean to not expose the whole wishlist type and create a loop

Reply:
Yes, I'd lean that way too. I suppose I could envision a scenario where the UI would want to show the names of the wishlists that the product belonged to, but I think it's perfectly reasonable to have the front end make an additional query for the full wishlist data.

As long as the minimal type would give us enough information to look up the full wishlist, I think that's fine 👍 .

For example, type gives `wishlistID` and then there's a way to get wishlists by `wishlistID`.